pkg/nsx/cluster.go

```diff
@@ -418,3 +425,83 @@ func (nsxVersion *NsxVersion) featureSupported(feature int) bool {
 	}
 	return false
 }
+
+func (cluster *Cluster) createHttpRequest(api string, ep *Endpoint) (*http.Request, error) {
+	return http.NewRequest("GET", fmt.Sprintf("%s://%s/%s", ep.Scheme(), ep.Host(), api), nil)
+}
+
+func (cluster *Cluster) getLicenseFromNsx() (*util.NsxLicense, error) {
+	ep := cluster.endpoints[0]
+	req, err := cluster.createHttpRequest("api/v1/licenses/licensed-features", ep)
+	if err != nil {
+		log.Error(err, "failed to create http request")
+		return nil, err
+	}
+	err = ep.UpdateHttpRequestAuth(req)
+	if err != nil {
+		log.Error(err, "keep alive update auth error")
+		return nil, err
+	}
+
+	resp, err := ep.noBalancerClient.Do(req)
+	if err != nil {
+		log.Error(err, "failed to get nsx license")
+		return nil, err
+	}
+	nsxLicense := &util.NsxLicense{}
+	err, _ = util.HandleHTTPResponse(resp, nsxLicense, true)
+	return nsxLicense, err
+}
+
+func (cluster *Cluster) getVersionWithRetries(delay time.Duration, maxRetry int) (*NsxVersion, error) {
+	var err error
+	for i := 0; i < maxRetry; i++ {
+		nsxVersion, err := cluster.GetVersion()
+		if err != nil {
+			log.Error(err, "failed to get nsx version")
+			rand, err := rand.Int(rand.Reader, big.NewInt(1000))
```

Comment:
just curious , why do we add a random number to delay?

Reply:
to avoid different routine sending the request at the same time.